src/MIDebugEngine/AD7.Impl/AD7Disassembly.cs

```diff
@@ -162,6 +176,28 @@ public int Read(uint dwInstructions, enum_DISASSEMBLY_STREAM_FIELDS dwFields, ou
                     }
                 }
 
+                if (!string.IsNullOrWhiteSpace(instruction.File))
+                {
+                    if ((dwFields & enum_DISASSEMBLY_STREAM_FIELDS.DSF_DOCUMENTURL) != 0)
+                    {
+                        prgDisassembly[iOp].dwFields |= enum_DISASSEMBLY_STREAM_FIELDS.DSF_DOCUMENTURL;
+                        prgDisassembly[iOp].bstrDocumentUrl = instruction.File;
+                    }
```

Comment:
Technically, we should also be returning this if it is different from the last instruction.